pkg/controllers/customcluster_controller.go

```diff
@@ -210,46 +231,170 @@ func (r *CustomClusterController) reconcile(ctx context.Context, customCluster *
 		return r.reconcileHandleProvisioning(ctx, customCluster, customMachine, cluster)
 	}
 
+	// If customCluster is in phase ProvisionedPhase, the controller will check whether to scale or do something else.
+	if phase == v1alpha1.ProvisionedPhase {
+		return r.reconcileProvisioned(ctx, customCluster, customMachine, cluster)
+	}
+
+	if phase == v1alpha1.ScalingUpPhase {
+		return r.reconcileHandleScalingUp(ctx, customCluster, customMachine, cluster)
```

Comment:
it looks weird, when phase is `ScalingUpPhase`, we enter `reconcileHandleScalingUp`

It should be when we enter `reconcileHandleScalingUp`, we setup the phase = reconcileHandleScalingUp